Refit/RefitResponse.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
+
+namespace Refit
+{
+    public class RefitResponse<T> : IRefitResponse<T>
```

Comment:
Please rename to `ApiResponse` for consistency with `ApiException`.

Reply:
I would suggest this be a `struct` and delegate the internal calls to the `HttpResponseMessage` instance.

Reply:
Finally, have a "ThrowIfNotSuccess" method on it that throws an `ApiException` just like it would have if it was a `T` response.

Reply:
@bennor renammed as suggested.
@onovotny changed to struct. I do not quite understand the other suggestions can give me some examples. I think you are asking for something like ApiResponse.Create(...,...); but T is ApiResponse<T> so pass T means ApiResponse<ApiResponse<T>>.Create(...,...).